### PR TITLE
Add "lease" section to vcd_vapp

### PR DESCRIPTION
**Pull request by dataclouder**
Add "lease" section to `vcd_vapp` resource and data source.
They work similarly to section "vapp_lease" in `vcd_org`. Using this section we can get information about the lease and eventually modify it.
When the "lease" block is not set, the vApp inherits the lease terms from the parent organization.

### CODE REVIEW

File: .changes/v3.5.0/762-improvements.md [hunk 1]
```diff
@@ -0,0 +1 @@
+* **Resource** and **Data source** `vcd_vapp` have a new `lease` block to view and change lease settings.
```
Comment (lvirbalas) [inline .changes/v3.5.0/762-improvements.md:1]:
```suggestion
* **Resource** and **Data source** `vcd_vapp` a new `lease` block to view and change lease settings.
```

File: vcd/datasource_vcd_vapp.go [hunk 2]
```diff
@@ -53,6 +53,25 @@ func datasourceVcdVApp() *schema.Resource {
...
+			"lease": &schema.Schema{
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Defines lease parameters for this vApp",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"runtime_lease_in_sec": &schema.Schema{
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "How long any of the VMs in the vApp can run before the vApp is automatically powered off or suspended.  0 means never expires",
```
Comment (lvirbalas) [inline vcd/datasource_vcd_vapp.go:65]:
```suggestion
							Description: "How long any of the VMs in the vApp can run before the vApp is automatically powered off or suspended. 0 means never expires",
```

**Review by lvirbalas — CHANGES_REQUESTED** [review]

File: .changes/v3.5.0/762-improvements.md (hunk 1 above)
Comment (dataclouder) [reply .changes/v3.5.0/762-improvements.md:1]: Why removing the verb? Now the object ("a new lease block") is hanging.

Comment (lvirbalas) [reply .changes/v3.5.0/762-improvements.md:1]: True, better transform it to the format used previously with "adds" word, few examples:

"datasource/vcd_resource_list adds vcd_right, vcd_role, vcd_global_role, vcd_rights_bundle to the supported resource types (#677)"
"resource/vcd_vapp_vm adds support to update storage_profile (#580)"

Comment (dataclouder) [reply .changes/v3.5.0/762-improvements.md]: Updated

File: vcd/datasource_vcd_vapp.go (hunk 2 above)
Comment (dataclouder) [reply vcd/datasource_vcd_vapp.go]: Updated

**Review by lvirbalas — APPROVED** [review]
Thanks!

**Review by vbauzysvmware — APPROVED** [review]

**Review by Didainius — APPROVED** [review]

**Pull request merged**